client/app/src/main/java/com/microsoft/research/karya/ui/registration/OTPFragment.kt

```diff
@@ -32,32 +30,42 @@ class OTPFragment : Fragment(R.layout.fragment_otp) {
     private lateinit var karyaAPI: KaryaAPIService
 
     private fun setupObservers() {
-        viewModel.openDashBoardFromOTP.observe(viewLifecycleOwner, Observer { openDashBoard ->
+        viewModel.openDashBoardFromOTP.observe(viewLifecycleOwner) { openDashBoard ->
             if (openDashBoard) {
                 navigateToDashBoard()
             }
-        })
+        }
 
-        viewModel.openProfilePictureFragmentFromOTP.observe(viewLifecycleOwner, { openProfilePictureFragment ->
+        viewModel.openProfilePictureFragmentFromOTP.observe(viewLifecycleOwner) { openProfilePictureFragment ->
             if (openProfilePictureFragment) {
                 navigateToProfilePicture()
             }
-        })
+        }
 
-        viewModel.currOtpVerifyState.observe(viewLifecycleOwner, { state ->
+        viewModel.currOtpVerifyState.observe(viewLifecycleOwner) { state ->
             when (state) {
                 OtpVerifyState.SUCCESS -> onOtpVerifySuccess()
                 OtpVerifyState.FAIL -> onOtpVerifyOrResendFailure()
                 OtpVerifyState.NOT_ENTERED -> setOtpNotSentUI()
             }
-        })
+        }
 
-        viewModel.currOtpResendState.observe(viewLifecycleOwner, { state ->
+        viewModel.currOtpResendState.observe(viewLifecycleOwner) { state ->
             when (state) {
                 // TODO: Maybe indicate user after successful otp sent
                 OtpSendState.FAIL -> onOtpVerifyOrResendFailure()
             }
-        })
+        }
+
+        viewModel.idTokenLiveData.observe(viewLifecycleOwner) { idToken ->
+            if (!idToken.isNullOrEmpty()) {
+                val sharedPref = requireActivity().getPreferences(Context.MODE_PRIVATE)
+                with(sharedPref.edit()) {
+                    putString(getString(R.string.id_token_key), idToken)
```

Comment:
Don't use `stringRes` for preference keys. We already have a `PreferenceKeys` class inside utils please use that. Also instead of using the `sharedPrefs` directly, use `DataStore` api. Check `SplashFragment` if you want to know how we're using `DataStore`.